### PR TITLE
python310Packages.pystardict: init at 0.8

### PR DESCRIPTION
**Pull request by thornycrackers**
###### Description of changes

Python Package for request https://github.com/NixOS/nixpkgs/issues/241696

After building the package I tested it by using the example demo file (https://github.com/lig/pystardict/blob/master/examples/demo.py) but with python3 syntax.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pystardict/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "lig";
+    repo = pname;
+    rev = version;
+    hash = "sha256-YrZpIhyxfA3G7rP0SJ+EvzGwAXlne80AYilkj6cIDnA=";
+  };
+
+  SETUPTOOLS_SCM_PRETEND_VERSION = version;
```
Comment (natsukium) [inline pkgs/development/python-modules/pystardict/default.nix:22]:
```suggestion
  env.SETUPTOOLS_SCM_PRETEND_VERSION = version;
```

File: pkgs/development/python-modules/pystardict/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  propagatedBuildInputs = [ six setuptools ];
```
Comment (natsukium) [inline pkgs/development/python-modules/pystardict/default.nix:24]: Is `setuptools` required for `propagatedBuildInputs`?

File: pkgs/development/python-modules/pystardict/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+  nativeCheckInputs = [ pytestCheckHook ];
+
+  pythonImportsCheck = [ "pystardict" ];
+
+  meta = with lib; {
+    description =
+      "Library for manipulating StarDict dictionaries from within Python";
+    homepage = "https://github.com/lig/pystardict";
+    license = licenses.gpl3Only;
```
Comment (natsukium) [inline pkgs/development/python-modules/pystardict/default.nix:36]:
```suggestion
    license = licenses.gpl3Plus;
```
see, https://github.com/lig/pystardict/blob/ce2bd7b4289411c2ac1536a4d186ade9c3a79aba/setup.py#L9-L10

File: pkgs/development/python-modules/pystardict/default.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "pystardict";
+  version = "0.8";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "lig";
+    repo = pname;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pystardict/default.nix:17]:
```suggestion
    repo = "pystardict";
```

File: pkgs/development/python-modules/pystardict/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/pystardict/default.nix:24]: I think setuptools is only required for building so it should be moved to nativeBuildInputs

**Review by SuperSandro2000 — COMMENTED** [review]
The version number is missing in the commit message and PR title.

File: pkgs/development/python-modules/pystardict/default.nix (hunk 1 above)
Comment (thornycrackers) [reply pkgs/development/python-modules/pystardict/default.nix]: Good catch

File: pkgs/development/python-modules/pystardict/default.nix (hunk 2 above)
Comment (thornycrackers) [reply pkgs/development/python-modules/pystardict/default.nix]: Updated.

File: pkgs/development/python-modules/pystardict/default.nix (hunk 3 above)
Comment (thornycrackers) [reply pkgs/development/python-modules/pystardict/default.nix]: Ahhh that's how you tell. Thanks for linking to direct part. Kept wondering what the difference was

File: pkgs/development/python-modules/pystardict/default.nix (hunk 4 above)
Comment (thornycrackers) [reply pkgs/development/python-modules/pystardict/default.nix]: Updated. I have a couple other ones that use pname in the repo, should I got back and fix those?

**Comment by thornycrackers** [comment]
Captured all the changes, rebuilt and tested locally. Thank you for reviewing.

**Pull request merged**